### PR TITLE
Rename module for consistency, add Canceler

### PR DESCRIPTION
**Pull request by garyb**

### CODE REVIEW

File: src/Control/Monad/Fork/Canceler.purs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+module Control.Monad.Fork.Canceler where
+
+import Prelude
+
+import Control.Monad.Eff.Exception (Error)
+import Data.Monoid (class Monoid)
+import Data.Newtype (class Newtype)
+
+-- | Semigroup (and monoid) for combining cancel functions produced by `fork`.
+newtype Canceler m = Canceler (Error → m Boolean)
```
Comment (natefaubion) [inline src/Control/Monad/Fork/Canceler.purs]: This doesn't work because the Error is abstract in MonadFork. Should be `Canceler e m`.

**Review by natefaubion — CHANGES_REQUESTED** [review]

File: src/Control/Monad/Fork/Canceler.purs (hunk 1 above)
Comment (garyb) [reply src/Control/Monad/Fork/Canceler.purs]: Excellent point.

File: src/Control/Monad/Fork/Class.purs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import Control.Monad.Reader.Trans (ReaderT(..))
+import Control.Monad.Trans.Class (lift)
+
+class Monad m ⇐ MonadFork e m | m → e where
+  fork ∷ ∀ a. m a → m (e → m Boolean)
+
+instance monadForkAff ∷ MonadFork Error (Aff.Aff eff) where
+  fork aff = do
+    ac ← Aff.forkAff aff
+    pure \reason → Aff.cancel ac reason
```
Comment (natefaubion) [inline src/Control/Monad/Fork/Class.purs]: I think this can be `Aff.cancel <$> Aff.forkAff aff`

Comment (natefaubion) [reply src/Control/Monad/Fork/Class.purs]: `map Aff.cancel <<< Aff.forkAff` 👀

**Review by natefaubion — APPROVED** [review]

**Pull request merged**